### PR TITLE
add ZTransducer#iso8859_1

### PR DESCRIPTION
**Pull request by mschuwalow**
Part of #3865

### CODE REVIEW

File: streams/shared/src/main/scala/zio/stream/ZTransducer.scala [hunk 1]
```diff
@@ -540,6 +540,18 @@ object ZTransducer {
         }
     }
 
+  /**
+   * Decodes chunks of ISO/IEC 8859-1 bytes into strings.
+   *
+   * This transducer uses the String constructor's behavior when handling malformed byte
+   * sequences.
+   */
+  val iso8859_1: ZTransducer[Any, Nothing, Byte, String] =
```
Comment (iravid) [inline streams/shared/src/main/scala/zio/stream/ZTransducer.scala:549]: shall we call it iso8859_1Decode to be aligned with utf8Decode? And allow for a potential encoding transducer in the future.

File: streams/shared/src/main/scala/zio/stream/ZTransducer.scala [hunk 2]
```diff
@@ -540,6 +540,18 @@ object ZTransducer {
...
+    ZTransducer.fromPush {
+      case Some(is) => ZIO.succeedNow(Chunk.single(new String(is.toArray, "ISO-8859-1")))
```
Comment (fsvehla) [inline streams/shared/src/main/scala/zio/stream/ZTransducer.scala:551]: Using `java.nio.charset.StandardCharsets.ISO_8859_1` might save an allocation here

File: streams/shared/src/main/scala/zio/stream/ZTransducer.scala (hunk 1 above)
Comment (fsvehla) [inline streams/shared/src/main/scala/zio/stream/ZTransducer.scala:549]: In the JVM the charset is referred to as 'ISO_8859_1', maybe we should add a `_` after iso

**Comment by iravid** [comment]
Nice work Maxim, thank you!

**Pull request merged**